pkg/cmd/issue/view/view.go

```diff
@@ -101,6 +107,11 @@ func viewRun(opts *ViewOptions) error {
 	if opts.IO.IsStdoutTTY() {
 		return printHumanIssuePreview(opts.IO, issue)
 	}
+
+	if opts.CommentsProvided {
+		return printRawIssueComments(opts.IO.Out, issue)
```

Comment:
When invoked in non-tty, I made the assumption that if the `--comments` flag is provided the user would like to just output comments information and skip the issue metadata and body.

Reply:
That's a good call! For now, I think it's best if scripts fetch issue bodies and its comment thread separately.